src/parsers/cdm_parser.jl

```diff
@@ -736,6 +699,71 @@ function services_csv_parser(rsv_raw::DataFrames.DataFrame,gen_raw::DataFrames.D
     return services_dict
 end
 
+"""Return a dictionary that includes a vector of parsed forecast dictionaries."""
+function forecast_csv_parser(ps_dict::Dict, data::Dict)
+    forecasts = Dict(
+        "forecasts" => Vector{Dict}(),
+        "loads" => Vector{Dict}(),
+    )
+
+    # A system can only contain forecasts with one resolution. Skip any that don't
+    # match the first one found.
+    # TODO DT: probably not what we want...
+    first_resolution = nothing
+
+    component_map = retrieve(ps_dict, "name")
+    for key in keys(data)
+        for (name, value) in data[key]
+            if name == "Load"
+                continue
+            end
+            # TODO DT: should we verify that the delta between each timestamp is the same?
+            if value isa DataFrames.AbstractDataFrame
+                len = size(value)[1]
+                @assert len > 2
+                resolution = value[:DateTime][2] - value[:DateTime][1]
+            else
+                len = length(value)
+                @assert len > 2
+                timestamps = TimeSeries.timestamp(value)
+                resolution = timestamps[2] - timestamps[1]
+            end
+
+            component = get(component_map, name, nothing)
+            if isnothing(component)
+                @error "$name is not in component_map, skipping forecast"
+                continue
+            end
+
+            forecast = Dict(
+                "component" => component,
+                "label" => key,
+                "data" => value,
+                "length" => len,
+                "resolution" => resolution,
+            )
+            @info "adding $key generator forecasts"
+            if isnothing(first_resolution)
+                first_resolution = resolution
+            elseif resolution != first_resolution
+                @error("found timeseries with a different resolution", resolution,
+                       first_resolution, maxlog=3)
+                continue
+            end
+            push!(forecasts["forecasts"], forecast)
+        end
+
+        # TODO DT: Is this used anywhere?
```

Comment:
The structure could be changed depending on how we will use this data.

Reply:
This looks good to me. I would just enable the possibility of having more than one forecast dict to accommodate more than one resolution when parsing the CSV's